cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/connection/ConnectionStore.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.cdap.datapipeline.connection;
+
+import com.google.gson.Gson;
+import io.cdap.cdap.api.NamespaceSummary;
+import io.cdap.cdap.api.dataset.lib.CloseableIterator;
+import io.cdap.cdap.spi.data.StructuredRow;
+import io.cdap.cdap.spi.data.StructuredTable;
+import io.cdap.cdap.spi.data.table.StructuredTableId;
+import io.cdap.cdap.spi.data.table.StructuredTableSpecification;
+import io.cdap.cdap.spi.data.table.field.Field;
+import io.cdap.cdap.spi.data.table.field.Fields;
+import io.cdap.cdap.spi.data.table.field.Range;
+import io.cdap.cdap.spi.data.transaction.TransactionRunner;
+import io.cdap.cdap.spi.data.transaction.TransactionRunners;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Store to save connection information.
+ *
+ *  The table will look like:
+ *  primary key                                                 column -> value
+ *  [namespace][connection-id]                                  connection_data -> Connection{@link Connection}
+ */
+public class ConnectionStore {
+  public static final StructuredTableId TABLE_ID = new StructuredTableId("connections");
+  private static final String NAMESPACE_FIELD = "namespace";
+  private static final String CONNECTION_ID_FIELD = "connection_id";
+  // this field is to ensure the connection information is correctly fetched if a namespace is recreated
+  private static final String GENERATION_COL = "generation";
+  private static final String CONNECTION_DATA_FIELD = "connection_data";
+
+  public static final StructuredTableSpecification CONNECTION_TABLE_SPEC =
+    new StructuredTableSpecification.Builder()
+      .withId(TABLE_ID)
+      .withFields(Fields.stringType(NAMESPACE_FIELD),
+                  Fields.longType(GENERATION_COL),
+                  Fields.stringType(CONNECTION_ID_FIELD),
+                  Fields.stringType(CONNECTION_DATA_FIELD))
```

Comment:
I saw in the mocks there is a "new" marker for newly added connections. Should we add columns for creation/update time?

Reply:
Yea just saw that, I added these two columns incase in feature we want to support sort by time